Description: Detect platforms Debian/kfreebsd and Debian/Hurd
Origin: http://anonscm.debian.org/gitweb/?p=pkg-games/bsnes.git;a=tree;f=debian/patches

--- a/bsnes/nall/intrinsics.hpp
+++ b/bsnes/nall/intrinsics.hpp
@@ -26,7 +26,7 @@
 
 /* Platform detection */
 
-#if defined(linux) || defined(__sun__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#if defined(linux) || defined(__sun__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__GNU__)
   #define PLATFORM_X
   Intrinsics::Platform Intrinsics::platform() { return Intrinsics::Platform::X; }
 #elif defined(__APPLE__)
--- a/snesfilter/nall/detect.hpp
+++ b/snesfilter/nall/detect.hpp
@@ -15,7 +15,7 @@
   #define PLATFORM_WIN
 #elif defined(__APPLE__)
   #define PLATFORM_OSX
-#elif defined(linux) || defined(__sun__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#elif defined(linux) || defined(__sun__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__GNU__)
   #define PLATFORM_X
 #endif
 
--- a/snespurify/nall/detect.hpp
+++ b/snespurify/nall/detect.hpp
@@ -15,7 +15,7 @@
   #define PLATFORM_WIN
 #elif defined(__APPLE__)
   #define PLATFORM_OSX
-#elif defined(linux) || defined(__sun__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
+#elif defined(linux) || defined(__sun__) || defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__GNU__)
   #define PLATFORM_X
 #endif
 
