Description: Include ALSA and udev only on Linux.
Author: Tobias Hansen <thansen@debian.org>

--- a/higan/target-tomoko/GNUmakefile
+++ b/higan/target-tomoko/GNUmakefile
@@ -27,8 +27,14 @@
   ruby += input.quartz input.carbon
 else ifeq ($(platform),linux)
   ruby += video.glx video.xvideo video.xshm video.sdl
-  ruby += audio.oss audio.alsa audio.openal audio.pulseaudio audio.pulseaudiosimple audio.ao
-  ruby += input.sdl input.xlib input.udev
+  ifneq ($(findstring Linux,$(uname)),)
+    ruby += audio.alsa
+  endif
+  ruby += audio.oss audio.openal audio.pulseaudio audio.pulseaudiosimple audio.ao
+  ifneq ($(findstring Linux,$(uname)),)
+    ruby += input.udev
+  endif
+  ruby += input.sdl input.xlib
 else ifeq ($(platform),bsd)
   ruby += video.glx video.xvideo video.xshm video.sdl
   ruby += audio.oss audio.openal
