--- a/bsnes/Makefile
+++ b/bsnes/Makefile
@@ -9,8 +9,8 @@
 # options += console
 
 # compiler
-c       := $(compiler) -std=gnu99
-cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
+c       := $(compiler) -std=gnu99 `dpkg-buildflags --get CFLAGS`
+cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x `dpkg-buildflags --get CPPFLAGS`
 flags   := -I. -O1 -fthread-jumps -falign-functions  -falign-jumps -falign-loops \
 -falign-labels -fcaller-saves -fcrossjumping -fcse-follow-jumps \
 -fcse-skip-blocks -fdelete-null-pointer-checks -fdevirtualize \
--- a/snesfilter/Makefile
+++ b/snesfilter/Makefile
@@ -1,7 +1,7 @@
 include nall/Makefile
 
-c       := $(compiler) -std=gnu99
-cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x
+c       := $(compiler) -std=gnu99 `dpkg-buildflags --get CFLAGS`
+cpp     := $(subst cc,++,$(compiler)) -std=gnu++0x `dpkg-buildflags --get CPPFLAGS`
 flags   := -fPIC -I. -Iobj -O3 -fomit-frame-pointer
 link    := -s
 objects :=
--- a/snespurify/cc-gtk.sh
+++ b/snespurify/cc-gtk.sh
@@ -1,4 +1,5 @@
-g++ -std=gnu++0x -I. -O1 -fthread-jumps -falign-functions  -falign-jumps -falign-loops \
+g++ -std=gnu++0x `dpkg-buildflags --get CPPFLAGS` -I. -O1 -fthread-jumps \
+-falign-functions  -falign-jumps -falign-loops \
 -falign-labels -fcaller-saves -fcrossjumping -fcse-follow-jumps \
 -fcse-skip-blocks -fdelete-null-pointer-checks -fdevirtualize \
 -fexpensive-optimizations -fgcse -fgcse-lm -finline-small-functions \
@@ -9,7 +10,8 @@
 -ftree-pre -ftree-vrp -finline-functions -funswitch-loops -fpredictive-commoning \
 -fgcse-after-reload -fipa-cp-clone -fomit-frame-pointer -c phoenix/phoenix.cpp \
 `pkg-config --cflags gtk+-2.0` -DPHOENIX_GTK
-g++ -std=gnu++0x -I. -O1 -fthread-jumps -falign-functions  -falign-jumps -falign-loops \
+g++ -std=gnu++0x `dpkg-buildflags --get CPPFLAGS` -I. -O1 -fthread-jumps \
+-falign-functions  -falign-jumps -falign-loops \
 -falign-labels -fcaller-saves -fcrossjumping -fcse-follow-jumps \
 -fcse-skip-blocks -fdelete-null-pointer-checks -fdevirtualize \
 -fexpensive-optimizations -fgcse -fgcse-lm -finline-small-functions \
