--- a/higan/target-tomoko/program/media.cpp
+++ b/higan/target-tomoko/program/media.cpp
@@ -16,7 +16,7 @@
 auto Program::loadMedia(Emulator::Interface& emulator_, Emulator::Interface::Media& media, const string& location) -> void {
   unloadMedia();
 
-  mediaPaths(0) = locate({localpath(), "higan/"}, {media.name, ".sys/"});
+  mediaPaths(0) = locate({sharedpath(), "higan/"}, {media.name, ".sys/"});
   mediaPaths(media.id) = location;
   folderPaths.append(location);
 
