From: David James <davidjamescastor215@proton.me>
Date: Sun, 31 May 2026 12:01:12 +0100
Subject: Use the system libchdr

---
 thirdparty/CMakeLists.txt | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/thirdparty/CMakeLists.txt b/thirdparty/CMakeLists.txt
index 238ed5d1e..f2b9979e9 100644
--- a/thirdparty/CMakeLists.txt
+++ b/thirdparty/CMakeLists.txt
@@ -33,8 +33,18 @@ target_include_directories(sevenzip-sdk PUBLIC lzma-sdk-26.02/C)
 set_target_properties(sevenzip-sdk PROPERTIES POSITION_INDEPENDENT_CODE ON FOLDER thirdparty PREFIX "")
 
 option(ARES_ENABLE_CHD "Enable CHD format support via libchdr" ON)
+option(WITH_SYSTEM_LIBCHDR "Use the system libchdr instead of bundled" OFF)
 
 if(ARES_ENABLE_CHD)
+  if(WITH_SYSTEM_LIBCHDR)
+    include(FindPkgConfig)
+    pkg_check_modules(CHDR REQUIRED IMPORTED_TARGET libchdr)
+    pkg_check_modules(LZMA REQUIRED IMPORTED_TARGET liblzma)
+    add_library(chdr-static INTERFACE)
+    add_library(lzma INTERFACE)
+    target_link_libraries(chdr-static INTERFACE PkgConfig::CHDR)
+    target_link_libraries(lzma INTERFACE PkgConfig::LZMA)
+  else()
   # lzma
   add_subdirectory(libchdr/deps/lzma-25.01 EXCLUDE_FROM_ALL)
   list(APPEND CHDR_INCLUDES lzma)
@@ -84,6 +94,7 @@ if(ARES_ENABLE_CHD)
     chdr-static
     PRIVATE $<$<CXX_COMPILER_ID:Clang,AppleClang>:-Wno-unreachable-code -Wno-unused-function>
   )
+  endif()
 endif()
 
 add_library(
@@ -136,7 +147,7 @@ add_library(qon STATIC qon/qon.cpp)
 set_target_properties(ymfm PROPERTIES FOLDER thirdparty PREFIX "")
 set_target_properties(tzxfile PROPERTIES FOLDER thirdparty PREFIX "")
 set_target_properties(sljit PROPERTIES FOLDER thirdparty PREFIX "")
-if(ARES_ENABLE_CHD)
+if(ARES_ENABLE_CHD AND NOT WITH_SYSTEM_LIBCHDR)
   set_target_properties(chdr-static PROPERTIES FOLDER thirdparty PREFIX "")
   set_target_properties(miniz PROPERTIES FOLDER thirdparty PREFIX "")
   set_target_properties(zstd PROPERTIES FOLDER thirdparty PREFIX "")
