From: David James <davidjamescastor215@proton.me>
Date: Sun, 31 May 2026 12:01:12 +0100
Subject: Use the system libchdr

---
 thirdparty/CMakeLists.txt | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/thirdparty/CMakeLists.txt b/thirdparty/CMakeLists.txt
index 614713d..ece0fca 100644
--- a/thirdparty/CMakeLists.txt
+++ b/thirdparty/CMakeLists.txt
@@ -5,8 +5,18 @@ target_compile_definitions(sljit PUBLIC SLJIT_HAVE_CONFIG_PRE=1 SLJIT_HAVE_CONFI
 target_compile_options(sljit PRIVATE $<$<COMPILE_LANG_AND_ID:C,AppleClang,Clang,GNU>:-Wno-conditional-uninitialized>)
 
 option(ARES_ENABLE_CHD "Enable CHD format support via libchdr" ON)
+option(WITH_SYSTEM_LIBCHDR "Use the system libchdr instead of bundled" OFF)
 
 if(ARES_ENABLE_CHD)
+if(WITH_SYSTEM_LIBCHDR)
+  include(FindPkgConfig)
+  pkg_check_modules(CHDR REQUIRED IMPORTED_TARGET libchdr)
+  pkg_check_modules(LZMA REQUIRED IMPORTED_TARGET liblzma)
+  add_library(chdr-static INTERFACE)
+  add_library(lzma INTERFACE)
+  target_link_libraries(chdr-static INTERFACE PkgConfig::CHDR)
+  target_link_libraries(lzma INTERFACE PkgConfig::LZMA)
+else()
   # lzma
   add_subdirectory(libchdr/deps/lzma-25.01 EXCLUDE_FROM_ALL)
   list(APPEND CHDR_INCLUDES lzma)
@@ -57,6 +67,7 @@ if(ARES_ENABLE_CHD)
     PRIVATE $<$<CXX_COMPILER_ID:Clang,AppleClang>:-Wno-unreachable-code -Wno-unused-function>
   )
 endif()
+endif()
 
 add_library(
   tzxfile
@@ -109,8 +120,10 @@ set_target_properties(ymfm PROPERTIES FOLDER thirdparty PREFIX "")
 set_target_properties(tzxfile PROPERTIES FOLDER thirdparty PREFIX "")
 set_target_properties(sljit PROPERTIES FOLDER thirdparty PREFIX "")
 if(ARES_ENABLE_CHD)
+if(NOT WITH_SYSTEM_LIBCHDR)
   set_target_properties(chdr-static PROPERTIES FOLDER thirdparty PREFIX "")
   set_target_properties(miniz PROPERTIES FOLDER thirdparty PREFIX "")
   set_target_properties(zstd PROPERTIES FOLDER thirdparty PREFIX "")
 endif()
+endif()
 set_target_properties(qon PROPERTIES FOLDER thirdparty PREFIX "")
