Description: Use the libchdr packaged separately
Author: Alexandre Detiste

diff --git a/thirdparty/GNUmakefile b/thirdparty/GNUmakefile
index 5390f9f3..5f3d1dfe 100644
--- a/thirdparty/GNUmakefile
+++ b/thirdparty/GNUmakefile
@@ -43,9 +43,9 @@ $(object.path)/zlib_trees.o: $(libchdr.path)/deps/zlib-1.2.13/trees.c
 $(object.path)/zlib_uncompr.o: $(libchdr.path)/deps/zlib-1.2.13/uncompr.c
 $(object.path)/zlib_zutil.o: $(libchdr.path)/deps/zlib-1.2.13/zutil.c
 
-flags += -I$(libchdr.path)/include
-flags += -I$(libchdr.path)/deps/lzma-19.00/include -D_7ZIP_ST
-flags += -I$(libchdr.path)/deps/zlib-1.2.13
+libchdr.objects :=
+libchdr.options := $(shell pkg-config --libs libchdr)
+flags += $(shell pkg-config --cflags libchdr) -D_7ZIP_ST
 
 # instruct glibc to declare fseeko/ftello
 flags += -D_LARGEFILE_SOURCE
